apollo-router/src/plugins/connectors/handle_responses.rs

```diff
@@ -36,7 +35,10 @@ pub(crate) enum HandleResponseError {
 // --- RESPONSES ---------------------------------------------------------------
 
 pub(crate) async fn handle_responses(
-    responses: Vec<http::Response<RouterBody>>,
+    responses: Vec<(
+        Result<http::Response<RouterBody>, ConnectorError>,
+        ResponseKey,
+    )>,
```

Comment:
Small nit, but I think that it might be worth making a struct to contain more complex tuples.

Reply:
There's additional refactoring that should be done here, but I held off on it for now. For instance, connectors shouldn't be tied to http at all, that entire body parameter should be generic. I'm looking at doing that as a follow-on.

Reply:
I went ahead and just refactored this here.